### PR TITLE
[INTLY-7222] Create a skeleton controller for metrics reconciliation in CRO

### PR DESCRIPTION
**Pull request by dimitraz**
## Overview

Work in progress 
Jira: https://issues.redhat.com/browse/INTLY-7222

## Verification

- Clone this branch
- Run `make cluster/prepare`
- Run `make run`
- Ensure that the new controller is registered in the operator logs 
- Ensure that the controller is being reconciled every 5 minutes

### CODE REVIEW

**Comment by aidenkeating** [comment]
👀

File: pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_cloudwatchmetrics")
+
+// Set the reconcile duration for this controller.
+// Currently it will be called once every 5 minutes
+const watchDuration = 5
```
Comment (aidenkeating) [inline pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go:23]: Can we replace this with seconds, just to allow for more accuracy

File: pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
+package cloudwatchmetrics
```
Comment (aidenkeating) [inline pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go:1]: Can we rename this to cloud resource metrics, or cloud metrics. To allow for it to be provider-independent.

We'll likely end up using this for GCP and Azure too.

File: pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	c, err := controller.New("cloudwatchmetrics-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Send a generic event to the channel to kick
+	// off the first reconcile
+	events := make(chan event.GenericEvent)
+	go func() {
+		events <- event.GenericEvent{
```
Comment (aidenkeating) [inline pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go:48]: Could we document at the top of this file the approach we're taking here.

- That this controller is for reconciling metrics for every resource (currently only redis and postgres)
- That this controller takes a sync the world approach
- That this controller will reconcile every set period of time

To make sure it's clear for future readers of the code

Comment (dimitraz) [reply pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go:48]: Sure

File: pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go (hunk 2 above)
Comment (dimitraz) [reply pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go:1]: Will do

File: pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go (hunk 1 above)
Comment (dimitraz) [reply pkg/controller/cloudwatchmetrics/cloudwatchmetrics_controller.go:23]: Ok :+1:

**Pull request merged**